FeedStoreChallenge/ObjectBoxFeedStore.swift

```
@@ -0,0 +1,161 @@
+	}
+	
+	// MARK: - Private properties
+	
+	private let storeURL: URL
+	private let store: Store
+	private let queue: DispatchQueue
+	
+	// MARK: - Public init & methods
+	
+	public init(storeURL: URL) throws {
+		self.storeURL = storeURL
+		self.store = try Store(directoryPath: storeURL.path)
+		self.queue = DispatchQueue(label: "\(type(of: self))", qos: .userInitiated, attributes: .concurrent)
+	}
+	
+	public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
+		queue.async(flags: .barrier) { [weak self] in
+			guard let self = self else { return }
+			
+			self.clearCache()
+			
+			completion(nil)
+		}
+	}
+	
+	public func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
+		queue.async(flags: .barrier) { [weak self] in
+			guard let self = self else { return }
+			let cache = Cache(timestamp: timestamp.timeIntervalSinceReferenceDate)
+			let storeFeeds: [StoreFeed] = feed.toStore(with: cache)
+			
+			self.clearCache()
+			self.setCache(storeFeeds: storeFeeds)
+			
+			completion(nil)
+		}
+	}
+	
+	public func retrieve(completion: @escaping RetrievalCompletion) {
+		queue.async { [weak self] in
+			guard let self = self else { return }
+			do {
+				guard let cache = self.getCache() else {
+					return completion(.empty)
+				}
+				let localFeedStore = try ObjectBoxFeedStore.map(cache.feed)
+				let dateTimestamp = Date(timeIntervalSinceReferenceDate: cache.timestamp)
+				
+				completion(.found(feed: localFeedStore, timestamp: dateTimestamp))
+			} catch {
+				completion(.failure(error))
+			}
+		}
+	}
+	
+	// MARK: - Private methods
+	
+	private func clearCache() {
+		removeAll(for: Cache.self)
+		removeAll(for: StoreFeed.self)
+	}
+	
+	@discardableResult
+	private func removeAll<T>(for entityType: T.Type = T.self) -> UInt64? where T : ObjectBox.EntityInspectable, T : ObjectBox.__EntityRelatable, T == T.EntityBindingType.EntityType {
+		return try? self.store.box(for: entityType).removeAll()
```

This operation can fail, but errors are ignored with `try?`.

Errors shouldn't be ignored. 

If there's an error, it should be thrown and the completion closure should be called with the error.

---

Yes, but here's the question: Since I couldn't made that method throw an exception (That doesn't throw when you try to perform any kind of operation from outside to the actual DB file, like deleting/overriding/changing permissions to it. I explained this in the commit history.) isn't just better to treat that method as if "doesn't throw", instead of adding a behaviour (complete with an error when get an exception) that could not be verified? (That will also decrease the global code coverage)

---

Hi @RRReDz, it's important to handle errors correctly - even if you can't simulate them during tests.